sale_crm_lead_product/models/sale_order.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (C) 2017-2024 ForgeFlow S.L. (https://www.forgeflow.com)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html)
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    show_copy_products_from_opportunity = fields.Boolean(
+        compute="_compute_show_copy_products_from_opportunity"
+    )
+
+    @api.depends("opportunity_id", "order_line")
```

Comment:
```suggestion
    @api.depends("opportunity_id", "order_line.crm_lead_line_id")
```
I think this could be better for performance